### PR TITLE
Fix a few issues with the plugin manifest

### PR DESCRIPTION
**Pull request by dkanada**
This removes the last of the annoying properties from the version and package info classes. The installation is now used internally to collect data from both classes, but at some point we could also start using a tuple instead.

### CODE REVIEW

**Comment by dkanada** [comment]
@mark-monteiro this should resolve the issue in #2951 by reverting the version to a string in the serialization object and parsing it as a version for the class that gets passed around for the install.

**Comment by dkanada** [comment]
Looks like I'll have to do another pass through web as well, this was mainly to remove the last few duplicate properties and fix the serialization issues.

**Comment by mark-monteiro** [comment]
> Looks like I'll have to do another pass through web as well, this was mainly to remove the last few duplicate properties and fix the serialization issues.

Yeah, the errors in the web client did not appear to be related to this PR. But it would be nice to have the web side working to test this.

**Comment by Artiume** [comment]
>  general this looks fine but I couldn't test because the web UI for installing plugins is broken. Is there a web PR that needs to go along with this?

@barronpm and I saw a similar issue in the User DB migrations for User Authentication

**Comment by barronpm** [comment]
The cause of the error @Artiume mentioned was accidentally adding the same assembly to the Enumerable returned by `ApplicationHost.GetComposablePartAssemblies`, but this doesn't look the same, as all of the options are "undefined" for you

**Comment by mark-monteiro** [comment]
I believe the issue I had was fixed here: https://github.com/jellyfin/jellyfin-web/pull/1291/files
Just need to find some time to come back to this to test it again

**Comment by dkanada** [comment]
@mark-monteiro there's still one more PR to merge in web updating the API client before this should be working again.

**Comment by dkanada** [comment]
Versions seem to be fine, but the installation is failing due to changes from the API client migration.

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 1]
```diff
@@ -330,50 +319,50 @@ public async Task InstallPackage(VersionInfo package, CancellationToken cancella
...
+                           ?? _applicationHost.Plugins.FirstOrDefault(p => p.Name.Equals(package.Name, StringComparison.OrdinalIgnoreCase));
 
             // Do the install
             await PerformPackageInstallation(package, cancellationToken).ConfigureAwait(false);
 
             // Do plugin-specific processing
             if (plugin == null)
             {
-                _logger.LogInformation("New plugin installed: {0} {1} {2}", package.name, package.version);
+                _logger.LogInformation("New plugin installed: {0} {1} {2}", package.Name, package.Version);
```
Comment (crobibero) [inline Emby.Server.Implementations/Updates/InstallationManager.cs:334]:
```suggestion
                _logger.LogInformation("New plugin installed: {0} {1}", package.Name, package.Version);
```

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 2]
```diff
@@ -330,50 +319,50 @@ public async Task InstallPackage(VersionInfo package, CancellationToken cancella
...
 
-                PluginInstalled?.Invoke(this, new GenericEventArgs<VersionInfo>(package));
+                PluginInstalled?.Invoke(this, package);
             }
             else
             {
-                _logger.LogInformation("Plugin updated: {0} {1} {2}", package.name, package.version);
+                _logger.LogInformation("Plugin updated: {0} {1} {2}", package.Name, package.Version);
```
Comment (crobibero) [inline Emby.Server.Implementations/Updates/InstallationManager.cs:340]:
```suggestion
                _logger.LogInformation("Plugin updated: {0} {1}", package.Name, package.Version);
```

**Comment by crobibero** [comment]
Looks like InstallationInfo.SourceUrl needs to be set in Emby.Server.Implementations\Updates\InstallationManager.cs L220

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 3]
```diff
@@ -211,52 +195,63 @@ public async Task<IReadOnlyList<PackageInfo>> GetAvailablePackages(CancellationT
...
+            var appVer = _applicationHost.ApplicationVersion;
+            var availableVersions = package.versions
+                .Where(x => Version.Parse(x.targetAbi) <= appVer);
+
+            if (minVersion != null)
+            {
+                availableVersions = availableVersions
+                    .Where(x => new Version(x.version) >= minVersion)
+                    .OrderByDescending(x => x.version);
             }
```
Comment (JustAMan) [inline Emby.Server.Implementations/Updates/InstallationManager.cs:210]:
```suggestion
            if (minVersion != null)
            {
                availableVersions = availableVersions.Where(x => new Version(x.version) >= minVersion);
            }
```

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 4]
```diff
@@ -211,52 +195,63 @@ public async Task<IReadOnlyList<PackageInfo>> GetAvailablePackages(CancellationT
...
 
-            return GetCompatibleVersions(
-                package.versions,
-                minVersion);
+            foreach (var v in availableVersions)
```
Comment (JustAMan) [inline Emby.Server.Implementations/Updates/InstallationManager.cs:212]:
```suggestion
            foreach (var v in availableVersions.OrderByDescending(x => x.version))
)
```

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 5]
```diff
@@ -211,52 +195,62 @@ public async Task<IReadOnlyList<PackageInfo>> GetAvailablePackages(CancellationT
...
+            if (minVersion != null)
+            {
+                availableVersions = availableVersions.Where(x => new Version(x.version) >= minVersion);
             }
 
-            return GetCompatibleVersions(
-                package.versions,
-                minVersion);
+            foreach (var v in availableVersions.OrderByDescending(x => x.version))
+)
```
Comment (crobibero) [inline Emby.Server.Implementations/Updates/InstallationManager.cs:211]:
```suggestion
```

File: Emby.Server.Implementations/Updates/InstallationManager.cs [hunk 6]
```diff
@@ -211,52 +195,62 @@ public async Task<IReadOnlyList<PackageInfo>> GetAvailablePackages(CancellationT
...
+            {
+                yield return new InstallationInfo
+                {
+                    Changelog = v.changelog,
+                    Guid = new Guid(package.guid),
+                    Name = package.name,
+                    Version = new Version(v.version)
```
Comment (crobibero) [inline Emby.Server.Implementations/Updates/InstallationManager.cs:218]:
```suggestion
                    Version = new Version(v.version),
                    SourceUrl = v.sourceUrl,
                    Checksum = v.checksum
```

**Comment by crobibero** [comment]
Publish failing due to following errors:

```
EntryPoints/ServerEventNotifier.cs(91,13): error CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
EntryPoints/ServerEventNotifier.cs(89,28): error CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
EntryPoints/ServerEventNotifier.cs(96,13): error CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
EntryPoints/ServerEventNotifier.cs(94,28): error CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
EntryPoints/ServerEventNotifier.cs(101,13): error CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
EntryPoints/ServerEventNotifier.cs(99,28): error CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
EntryPoints/ServerEventNotifier.cs(121,13): error CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
EntryPoints/ServerEventNotifier.cs(119,28): error CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [/home/vsts/work/1/s/Emby.Server.Implementations/Emby.Server.Implementations.csproj]
```

**Comment by dkanada** [comment]
I see you've been enabling more errors :laughing: I'll fix them now.

**Pull request merged**